src/libopensc/dir.c

```diff
@@ -83,6 +84,25 @@ parse_dir_record(sc_card_t *card, u8 ** buf, size_t *buflen, int rec_nr)
 		LOG_FUNC_RETURN(ctx, r);
 	LOG_TEST_RET(ctx, r, "EF(DIR) parsing failed");
 
+	conf_block = sc_get_conf_block(ctx, "framework", "pkcs15", 1);
+	if (conf_block)   {
+		scconf_block **blocks = NULL;
+		char aid_str[SC_MAX_AID_STRING_SIZE];
+		int ignore_app = 0;
+
+		sc_bin_to_hex(aid.value, aid.len, aid_str, sizeof(aid_str), 0);
+		blocks = scconf_find_blocks(card->ctx->conf, conf_block, "application", aid_str);
+		if (blocks)   {
+			ignore_app = (blocks[0] && scconf_get_str(blocks[0], "disable", 0));
+                        free(blocks);
+		 }
+
+		if (ignore_app)   {
+			sc_log(ctx, "Application '%s' ignored", aid_str);
```

Comment:
the length to be printed should be limited here, either by using `%.s` or by setting `aid_str[SC_MAX_AID_STRING_SIZE-1] = '\0'`